src/apps/aave/shortcuts.ts

```diff
@@ -216,6 +219,72 @@ const hook: ShortcutsHook = {
           ]
         },
       }),
+      createShortcut({
+        id: 'swap-deposit',
+        name: 'Swap & Deposit',
+        description: 'Swap assets and lend them to earn interest',
+        networkIds: [networkId],
+        category: 'swap-deposit',
+        triggerInputShape: {
+          swapFromToken: tokenAmountWithMetadata,
+          // set via shortcutTriggerArgs, the deposit token's address
+          tokenAddress: ZodAddressLowerCased,
+        },
+        async onTrigger({ swapFromToken, tokenAddress, address, networkId }) {
+          const walletAddress = address as Address
+          // use a placeholder amount non zero amount so tx simulation can succeed.
+          // squid postHook will replace this with the actual amount after swap.
+          const amount = 1n
+          const approveData = encodeFunctionData({
+            abi: erc20Abi,
+            functionName: 'approve',
+            args: [poolContractAddress, amount],
+          })
+          const supplyData = encodeFunctionData({
+            abi: poolV3Abi,
+            functionName: 'supply',
+            args: [tokenAddress, amount, walletAddress, 0],
+          })
+
+          return await prepareSwapTransactions({
+            networkId,
+            swapFromToken,
+            swapToAddress: tokenAddress,
+            walletAddress,
+            simulatedGasPadding: [0n, SIMULATED_DEPOSIT_GAS_PADDING],
+            postHook: {
```

Comment:
can you add a comment linking to the squid docs for this?  on first read I had questions that might be most easily handled w/ docs link.